src/entries/popup/hooks/approveAppRequest/useApproveAppRequestValidations.ts

```diff
@@ -17,8 +17,14 @@ export const useApproveAppRequestValidations = ({
   chainId: ChainId;
   selectedGas?: GasFeeParams | GasFeeLegacyParams;
 }) => {
-  const { connectedToHardhat } = useConnectedToHardhatStore();
-  const chainIdToUse = connectedToHardhat ? ChainId.mainnet : chainId;
+  const { connectedToHardhat, connectedToHardhatOp } =
+    useConnectedToHardhatStore.getState();
```

Comment:
`getState` is not needed here

Reply:
using `getState` would get the state at the time the hook was called, but won't react to changes
using the hook `useConnectedToHardhatStore` will make all dependent components rerender when this state changes